CMakeLists.txt

```diff
@@ -1,13 +1,92 @@
+                set(HIP_CLANG_PATH "${ROCM_PATH}/llvm/bin" CACHE PATH "Path to HIP clang binaries")
+            else()
+                set(HIP_CLANG_PATH $ENV{HIP_CLANG_PATH} CACHE PATH "Path to HIP clang binaries")
+            endif()
+        endif()
+
+        # note: could probably grab this path directly using hipconfig?
+        if (NOT DEFINED HIP_CLANG_INCLUDE_PATH)
+            if(NOT DEFINED ENV{HIP_CLANG_INCLUDE_PATH})
+                # probably need a better way to get the compiler version.. this will cause non-existent paths for non-clang compilers
+                set(HIP_CLANG_INCLUDE_PATH "${HIP_CLANG_PATH}/../lib/clang/${CMAKE_CXX_COMPILER_VERSION}/include" CACHE PATH "Path to HIP clang include directory")
+            else()
+                set(HIP_CLANG_INCLUDE_PATH $ENV{HIP_CLANG_INCLUDE_PATH} CACHE PATH "Path to HIP clang include directory")
+            endif()
+        endif()
+
+        # try to find hipconfig executable which can help detect platforms and include dirs
+        find_program(HIPCONFIG_PATH hipconfig HINTS "${HIP_PATH}/bin")
+        if(HIPCONFIG_PATH)
+            execute_process(COMMAND ${HIPCONFIG_PATH} --platform OUTPUT_VARIABLE KRONMULT_HIP_PLATFORM)
+        elseif(DEFINED ENV{HIP_PLATFORM})
+            set(KRONMULT_HIP_PLATFORM "$ENV{HIP_PLATFORM}")
+        else()
+            message(FATAL_ERROR "Could not determine HIP platform, make sure HIP_PLATFORM is set")
+        endif()
+
+        message(STATUS "HIP platform has been detected as ${KRONMULT_HIP_PLATFORM}")
+        # hip >= 4.2 is now using "amd" to identify platform
+        if(KRONMULT_HIP_PLATFORM STREQUAL "hcc" OR KRONMULT_HIP_PLATFORM STREQUAL "amd")
+            set(KRONMULT_PLATFORM_AMD 1)
+        # hip <= 4.1 uses "nvcc" to identify nvidia platforms, >= 4.2 uses "nvidia"
+        elseif(KRONMULT_HIP_PLATFORM STREQUAL "nvcc" OR KRONMULT_HIP_PLATFORM STREQUAL "nvidia")
+            set(KRONMULT_PLATFORM_NVCC 1)
+        endif()
+
+        set(HIP_VERBOSE_BUILD ON CACHE STRING "Verbose compilation for HIP")
+
+        # look for HIP cmake configs in different locations
+        list(APPEND CMAKE_MODULE_PATH "${HIP_PATH}/cmake" "${HIP_PATH}/lib/cmake" "${ROCM_PATH}")
+        list(APPEND CMAKE_PREFIX_PATH "${HIP_PATH}/cmake" "${HIP_PATH}/lib/cmake" "${ROCM_PATH}")
+        set(CMAKE_HIP_ARCHITECTURES OFF)
+        find_package(HIP REQUIRED 4.0)
+        if(KRONMULT_PLATFORM_AMD)
+            enable_language (HIP)
+            set_target_properties(hip-lang::device PROPERTIES INTERFACE_INCLUDE_DIRECTORIES ${HIP_PATH}/include)
+        endif()
+
+        message(STATUS "HIP PLATFORM: ${HIP_PLATFORM}")
+        message(STATUS "HIP COMPILER: ${HIP_COMPILER}")
+        message(STATUS "HIP RUNTIME: ${HIP_RUNTIME}")
+        message(STATUS "HIP Includes: ${HIP_INCLUDE_DIRS}")
+        message(STATUS "HIP Libraries: ${HIP_LIBRARIES}")
+
+        if(KRONMULT_PLATFORM_NVCC)
+            find_package(CUDA 9.0 REQUIRED)
```

Comment:
This is a step back from  `enable_language (CUDA)`. Can we stay with first-class language support?